relay-event-normalization/src/schema.rs

```diff
@@ -53,12 +81,66 @@ impl Processor for SchemaProcessor {
         meta: &mut Meta,
         state: &ProcessingState<'_>,
     ) -> ProcessingResult {
-        if value.is_none() && state.attrs().required && !meta.has_errors() {
-            meta.add_error(ErrorKind::MissingAttribute);
+        match self.required {
+            RequiredMode::DeleteParent => {
+                self.stack.push(SchemaState::default());
+            }
+            RequiredMode::DeleteValue => {
+                if value.is_none() && state.attrs().required && !meta.has_errors() {
+                    meta.add_error(ErrorKind::MissingAttribute);
+                }
+            }
         }
 
         Ok(())
     }
+
+    fn after_process<T: ProcessValue>(
+        &mut self,
+        value: Option<&T>,
+        meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        // The stack is not filled if we're in delete value mode.
+        let Some(current) = self.stack.pop() else {
+            debug_assert!(
+                matches!(self.required, RequiredMode::DeleteValue),
+                "processing stack should always have a value"
+            );
+
+            return Ok(());
+        };
+
+        // There is a require validation if the field is required and the value is `None`, or
+        // the current object had any required validations already.
+        let is_required_violation =
+            state.attrs().required && (value.is_none() || current.has_required_violation);
+
+        if let Some(container) = self.stack.last_mut() {
```

Comment:
nit: Since the mode is called `DeleteParent`.

```suggestion
        if let Some(parent) = self.stack.last_mut() {
```